### PR TITLE
Add a checkbox to allow a user to allow/disallow access to private Gists

### LINKED ISSUE #45: Add a checkbox that allows users to decide whether they want to add the`gist` scope
In order to read private Gists, we have to ask for the `gist` scope from GitHub. But some people don't trust us enough to give us write access to their gists (and GitHub has not separated "read private gists" from "write to gists", annoyingly), and others don't want reminders on their private gists.

It'd be nice if there were a checkbox/radio that allow signing-up users to define which scope they want. We don't need to do the work to add it for current users; they can just delete their account and re-sign-up (unless it's super easy to add to current users, which I doubt).

### PR DESCRIPTION
**Pull request by jasonlbeggs**
This should be good to go. Let me know if you'd like me to change anything. I tested it with valet share and it seems to work as planned.

Fixes #45.

### CODE REVIEW

File: app/Http/Controllers/Auth/AuthController.php [hunk 1]
```diff
@@ -32,8 +32,10 @@ public function __construct()
      */
     public function redirectToProvider()
     {
+        $scopes = request('allow-private') ? ['gist', 'user-email'] : ['user-email'];
```
[the remark this answers is not in the document]
Comment (jasonlbeggs) [reply app/Http/Controllers/Auth/AuthController.php]: Yep, you're right. I've pushed the fixed. Sorry about that.

**Comment by jakebathman** [comment]
Sweet! Matt will take a look too and merge once he's good.

File: resources/assets/sass/app.scss [hunk 2]
```diff
@@ -108,6 +108,18 @@ img {
   }
 }
 
+.checkbox-label {
+    display: block;
+    font-size: .75em;
```
Comment (mattstauffer) [inline resources/assets/sass/app.scss]: Could you put a `0` here--so `0.75em`? Also on lines 115 and 118. I just think CSS values read a little more clearly with the preceding 0.

Comment (jasonlbeggs) [reply resources/assets/sass/app.scss]: You're right. Prettier usually does that for me, but I disabled it so it wouldn't reformat the whole file. Thanks for the suggestion!

**Comment by mattstauffer** [comment]
SWEEET. Gonna go check it out. Thanks for this @jasonlbeggs!

**Pull request merged**